File: aula14/c.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n, w;
    scanf("%d", &n);
    scanf("%d", &w);
    int bears[n];
    int elephant[w];
    for (int i = 0; i < n; i++){
        scanf("%d", &bears[i]);
    }
    for (int i = 0; i < w; i++){
        scanf("%d", &elephant[i]);
    }
    int *menor;
    menor = min_element(elephant, elephant + w);
    printf("%d\n", *menor);
    for (int i = 0; i < w; i++){
        elephant[i] = elephant[i] - *menor;
    }
    for (int i = 0; i < w; i++){
        printf("%d ", elephant[i]);
    }
    printf("\n");
    int numero;
    numero = 0;
    for (int i = 0; i < n - w + 1; i++){
        int contem = 1;
        for (int j = 0; j < w - 1; j++){
            if(!(bears[i + j] >= elephant[j] && ((bears[i + j + 1] - bears[i + j]) == (elephant[j + 1] - elephant[j])) && bears[i + j + 1] >= elephant[j + 1])){
                contem = 0;
                break;
            }
        }
        if(contem){
            numero++;
        }
    }
    printf("%d\n", numero);
    return 0;
}

File: aula14/c.py
n, w = [int(i) for i in input().split()]
bears = [int(i) for i in input().split()]
elephant = [int(i) for i in input().split()]

la = True

if w > n:
    print(0)
    la = False

if w == n:
    if bears == elephant:
        print(1)
    else:
        print(0)
    la = False

if la:
    menor = min(elephant)
    index = elephant.index(menor)

    for i in range(w):
        elephant[i] -= menor

    numero = 0

    for i in range(0, n - w + 1, 1):
        vetor = bears[i:i+w]
        
        menorVetor = vetor[index]
        
        for i in range(w):
            vetor[i] -= menorVetor
        
        if elephant == vetor:
            numero += 1
    print(numero)